packages/api/endpoints/executions.js

```diff
@@ -79,6 +85,29 @@ async function del(req, res) {
   return res.send({ message: 'Record deleted' });
 }
 
+/**
+ * Get execution history for a single granule or multiple granules
+ *
+ * @param {Object} req - express request object
+ * @param {Object} res - express response object
+ * @returns {Promise<Object>} the promise of express response object
+ */
+async function searchByGranules(req, res) {
+  const payload = req.body;
+  const knex = await getKnexClient();
+  const granules = await getGranulesForPayload(payload, knex);
```

Comment:
If I'm understanding correctly, this will accept or return an array of granules like `{ granuleId: string, collectionId: string}`. And thus we have to look up the collection_cumulus_id for that collection in order to find the executions in PG.

So my question is: can this endpoint actually accept an array of granules like `{ granule_id: string, collection_cumulus_id: number}` as **input**? The answer really depends on when this will be merged/released. If it's being released in the next sprint release after merge, then the answer is probably no. But if it's released with the changes where the granule endpoint will be returning granules from PG with those properties, then the answer might be yes

It's probably a "no", but just checking to see if we can save some DB queries

Reply:
Yeah this is a feature that operators have been waiting on since last PI so I think the plan is to release it asap. Otherwise, I do get what you're saying here. Do you think it's worth adding a ticket to follow up once all the RDS stuff is out there, maybe for an IP sprint? We'd probably have to keep it backwards compatible, but we could add additional handling for the collection cumulus id. At the very least, then any request from the dashboard could save those DB queries.

Reply:
I would probably just hold off on a ticket because I recall now that the plan is for us to return the structure of returned API endpoints. There isn't even any planned work afaik to update the granule endpoints to return the data structure from Postgres vs the current Dynamo output structure.

So no work or follow up needed here. Thanks for clarifying